CMakeLists.txt

```diff
@@ -97,6 +99,37 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSIO
   # This is needed because the boost headers throw this warning.
   add_compile_options(-Wno-error=tautological-overlap-compare)
 endif()
+if (LIBCXX_VERSION VERSION_GREATER 10 AND IRODS_CXX_STANDARD EQUAL 17)
+  # C++17 deprecated the explicit template specialization std::allocator<void>
+  # and the two-argument version of std::allocator::allocate
+  # Starting with 11.0, libc++ enforces these deprecations, which causes
+  # warnings from the boost headers.
+  # Boost has removed references to the two-argument version of
+  # std::allocator::allocate in 1.75, which would solve the problem, but,
+  # while the template specification deprecation is not supposed to affect
+  # existing code, libc++ 11 causes warnings for references to
+  # std::allocator<void>. Maybe a future release of libc++ will fix this.
+  # Should we ever switch to C++20 (or newer), std::allocator<void> problem
+  # will go away, as the explicit template specialization is removed in C++20.
+  #
+  # In the meantime, we have a couple of options for dealing with this.
+  # 1. Define the preprocessor variable _LIBCPP_DISABLE_DEPRECATION_WARNINGS.
+  #    This will silence all deprecation warnings from libc++. References to
+  #    deprecated declarations from other sources will still cause a warning
+  #    and hault the build. This solution/workaround could potentially silence
```

Comment:
hault -> halt

Reply:
I have just now realized that I have probably been spelling this word wrong my entire life.

Reply:
amazing.